### PR TITLE
v0.11.0 Change the interface for signing oracle prices

### PR DESCRIPTION
**Pull request by Kenadia**
Two breaking changes:
* require a market or asset name (instead of an asset ID or asset name)
* require a human-readable price (automatically rounded to 18 decimals places with RoundHalfEven)

### CODE REVIEW

**Comment by Kenadia** [comment]
Rebased

**Comment by Kenadia** [comment]
Merged master

File: src/signable/oracle-price.ts [hunk 1]
```diff
@@ -1,53 +1,79 @@
...
-      .iushln(ORACLE_PRICE_FIELD_BIT_LENGTHS.oracleName)
-      .iadd(oracleNameBn);
+    const expirationEpochSeconds = isoTimestampToEpochSeconds(params.isoTimestamp);
 
     return new SignableOraclePrice({
-      signedAssetId: signedAssetId.toString(16),
-      price: params.price,
-      isoTimestamp: params.isoTimestamp,
+      signedAssetId,
+      signedPrice: signedPrice.round(RoundingMode.RoundHalfEven).toFixed(0),
```
Comment (BrendanChou) [inline src/signable/oracle-price.ts:69]: nit: why round instead of expecting it to already be an integer?

**Review by BrendanChou — APPROVED** [review]

File: src/signable/oracle-price.ts (hunk 1 above)
Comment (Kenadia) [reply src/signable/oracle-price.ts:69]: I was on the fence before, but think you're right it should throw instead

Comment (Kenadia) [reply src/signable/oracle-price.ts]: I was on the fence before, but think you're right it should throw instead

**Pull request merged**